sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -51,6 +52,23 @@ public virtual bool IsRetriable(HttpMessage message, Exception exception)
                    (exception is OperationCanceledException && !message.CancellationToken.IsCancellationRequested);
         }
 
+        internal bool IsError(HttpMessage message)
+        {
+            if (TryClassify(message, out bool isError))
+            {
+                return isError;
+            }
+
+            return IsErrorResponse(message);
+        }
+
+        /// <inheritdoc/>
+        public override bool TryClassify(HttpMessage message, out bool isError)
```

Comment:
Should this be: a) sealed, b) return true, c) set isError to whatever IsError returns, d) implement the table lookup logc in IsError?

This way users would not be able to override this method, start returning false, and thus make this ResponseClassifier not usable at the end of a chain.

Reply:
I like it.  Let me think.